internal/jimm/jwtgenerator/jwtgenerator.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2024 Canonical.
+
+// jwtgenerator generates JWT tokens to authenticate
+// and authorize messages to Juju controllers.
+// This package is more specialised than a generic
+// JWT token generator as it crafts Juju specific
+// permissions that are added as claims to the JWT
+// and therefore exists in JIMM's business logic layer.
+package jwtgenerator
+
+import (
+	"context"
+	"sync"
+
+	"github.com/juju/names/v5"
+	"github.com/juju/zaputil/zapctx"
+	"go.uber.org/zap"
+
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/jimmjwx"
+	"github.com/canonical/jimm/v3/internal/openfga"
+)
+
+// jwtGeneratorDatabase specifies the database interface used by the
+// JWT generator.
+type jwtGeneratorDatabase interface {
```

Comment:
i think these three interfaces should be exported as they are parameter types for the exported constructor

Reply:
also interface names do not need the `jwt` prefix now

Reply:
Will remove the jwt prefix, but why do they need to be exported? Maybe for a caller to verify that a struct satisfies the interface with a `_ = jwtGeneratorDatabase` but even that's a stretch because the caller can just call `New()` with whatever struct it has to satisfy the interface.

Reply:
but the thing is that you don't know what interface you need to satisfy without looking at the code since it is unexported. generally having exported methods with unexported argument types or returns is an anti-pattern

Reply:
>but the thing is that you don't know what interface you need to satisfy without looking at the code since it is unexported

But even if it was exported, how would you know how to satisfy the interface without looking at the code?